File: src/Rcm.Web/wwwroot/js/Aggregates/MeasurementAggregates.ts
export interface MeasurementAggregates
{
    readonly temperature:Aggregates;
    readonly pressure:Aggregates;
    readonly humidity:Aggregates;
}

export interface Aggregates
{
    readonly first:AggregateEntry;
    readonly min:AggregateEntry;
    readonly max:AggregateEntry;
    readonly last:AggregateEntry;
}

export interface AggregateEntry
{
    readonly time:Date;
    readonly value:number;
}

File: src/Device/Rcm.Device.Aggregates.Api/MeasurementAggregates.cs
namespace Rcm.Device.Aggregates.Api;

public class MeasurementAggregates
{
    public Aggregates Temperature { get; }
    public Aggregates Pressure { get; }
    public Aggregates Humidity { get; }

    public MeasurementAggregates(Aggregates temperature, Aggregates pressure, Aggregates humidity)
    {
        Temperature = temperature;
        Pressure = pressure;
        Humidity = humidity;
    }
}